### PR TITLE
Install query `.scm` files along parser libraries

### LINKED ISSUE #3931: Should Makefile install queries?
### Problem

The Makefile created by `tree-sitter generate` does nothing with the `queries/*.scm` files.  However, my understanding is that these files are likely to be wanted for common use-cases such as syntax highlighting?

I notice that this is in contrast to the Python and Rust bindings, which both include them.

### Expected behavior

Perhaps these files could be installed to, say, `/usr/share/tree-sitter/LANG/queries/`.

### PR DESCRIPTION
**Pull request by mavit**
Fixes #3931.

### CODE REVIEW

File: cli/src/templates/cmakelists.cmake [hunk 1]
```diff
@@ -53,6 +53,10 @@ install(FILES "${CMAKE_CURRENT_BINARY_DIR}/tree-sitter-PARSER_NAME.pc"
 install(TARGETS tree-sitter-PARSER_NAME
         LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 
+file(GLOB QUERIES queries/*.scm)
+install(FILES ${QUERIES}
+        DESTINATION "${CMAKE_INSTALL_DATADIR}/tree_sitter/tree-sitter-PARSER_NAME/queries")
```
Comment (ObserverOfTime) [inline cli/src/templates/cmakelists.cmake:58]:
```suggestion
        DESTINATION "${CMAKE_INSTALL_DATADIR}/tree-sitter/queries/PARSER_NAME")
```

File: cli/src/templates/makefile [hunk 2]
```diff
@@ -69,13 +70,14 @@ $(PARSER): $(SRC_DIR)/grammar.json
 	$(TS) generate $^
 
 install: all
-	install -d '$(DESTDIR)$(INCLUDEDIR)'/tree_sitter '$(DESTDIR)$(PCLIBDIR)' '$(DESTDIR)$(LIBDIR)'
+	install -d '$(DESTDIR)$(DATADIR)'/tree_sitter/'$(LANGUAGE_NAME)'/queries '$(DESTDIR)$(INCLUDEDIR)'/tree_sitter '$(DESTDIR)$(PCLIBDIR)' '$(DESTDIR)$(LIBDIR)'
```
Comment (ObserverOfTime) [inline cli/src/templates/makefile:73]:
```suggestion
	install -d '$(DESTDIR)$(DATADIR)'/tree-sitter/queries/PARSER_NAME '$(DESTDIR)$(INCLUDEDIR)'/tree_sitter '$(DESTDIR)$(PCLIBDIR)' '$(DESTDIR)$(LIBDIR)'
```

File: cli/src/templates/makefile [hunk 3]
```diff
@@ -69,13 +70,14 @@ $(PARSER): $(SRC_DIR)/grammar.json
...
 	install -m644 bindings/c/$(LANGUAGE_NAME).h '$(DESTDIR)$(INCLUDEDIR)'/tree_sitter/$(LANGUAGE_NAME).h
 	install -m644 $(LANGUAGE_NAME).pc '$(DESTDIR)$(PCLIBDIR)'/$(LANGUAGE_NAME).pc
 	install -m644 lib$(LANGUAGE_NAME).a '$(DESTDIR)$(LIBDIR)'/lib$(LANGUAGE_NAME).a
 	install -m755 lib$(LANGUAGE_NAME).$(SOEXT) '$(DESTDIR)$(LIBDIR)'/lib$(LANGUAGE_NAME).$(SOEXTVER)
 	ln -sf lib$(LANGUAGE_NAME).$(SOEXTVER) '$(DESTDIR)$(LIBDIR)'/lib$(LANGUAGE_NAME).$(SOEXTVER_MAJOR)
 	ln -sf lib$(LANGUAGE_NAME).$(SOEXTVER_MAJOR) '$(DESTDIR)$(LIBDIR)'/lib$(LANGUAGE_NAME).$(SOEXT)
+	install -m644 queries/*.scm '$(DESTDIR)$(DATADIR)'/tree_sitter/'$(LANGUAGE_NAME)'/queries/
```
Comment (ObserverOfTime) [inline cli/src/templates/makefile:80]:
```suggestion
	install -m644 queries/*.scm '$(DESTDIR)$(DATADIR)'/tree-sitter/queries/PARSER_NAME
```

File: cli/src/templates/makefile [hunk 4]
```diff
@@ -69,13 +70,14 @@ $(PARSER): $(SRC_DIR)/grammar.json
...
 
 uninstall:
 	$(RM) '$(DESTDIR)$(LIBDIR)'/lib$(LANGUAGE_NAME).a \
```
Comment (ObserverOfTime) [inline cli/src/templates/makefile:83]: Don't forget to uninstall them.
```make
$(RM) -r '$(DESTDIR)$(DATADIR)'/tree-sitter/queries/PARSER_NAME
```

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]

**Review by GitbenmL — APPROVED** [review]

File: cli/src/templates/cmakelists.cmake (hunk 1 above)
Comment (mavit) [reply cli/src/templates/cmakelists.cmake]: Works for me.  I admit I vacillated between these two possible arrangements before suggesting the one I originally did.

**Review by amaanq — APPROVED** [review]

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**